packages/dds/sequence/src/test/sharedStringWithSequentialId.spec.ts

```diff
@@ -0,0 +1,117 @@
+                });
+
+    function validateMarker(marker: Marker): void {
+        const id = marker.getId();
+        assert(id !== undefined && id.length > 0, "Id should be defined and have a valid length");
+        assert(isValidSeqId(id), "Id is a valid sequential id");
+
+        // Validate summary
+        const segmentJson = marker.toJSONObject();
+        assert(segmentJson.props !== undefined && segmentJson.props[reservedMarkerIdKey] === id);
+    }
+
+    it("should have a marker id after marker insertion", async ()=> {
+        sharedStringWithSequentialId.insertMarker(0, ReferenceType.Tile);
+        const segment = sharedStringWithSequentialId.getContainingSegment(0).segment;
+        // Process the message.
+        containerRuntimeFactory.processAllMessages();
+
+        assert(Marker.is(segment), "Make sure we found our marker");
+        validateMarker(segment);
+    });
+
+    it("should have a marker id after marker insertion after existing marker", async ()=> {
+        sharedStringWithSequentialId.insertMarker(0, ReferenceType.Tile);
+        containerRuntimeFactory.processAllMessages();
+        sharedStringWithSequentialId.insertMarker(1, ReferenceType.Tile);
+
+        // Process all messages.
+        containerRuntimeFactory.processAllMessages();
+
+        const segmentAt0 = sharedStringWithSequentialId.getContainingSegment(0).segment;
+        const segmentAt1 = sharedStringWithSequentialId.getContainingSegment(1).segment;
+
+        assert(Marker.is(segmentAt0) && Marker.is(segmentAt1), "Make sure we found markers");
+        validateMarker(segmentAt0);
+        validateMarker(segmentAt1);
+        const id0 = segmentAt0.getId();
+        const id1 = segmentAt1.getId();
+        assert(id0 < id1, "Id at 0 should be smaller than id at 1");
+    });
+
+    it("should have a marker id after marker insertion before existing marker", async ()=> {
+        sharedStringWithSequentialId.insertMarker(0, ReferenceType.Tile, { insert1: true });
+        containerRuntimeFactory.processAllMessages();
+        sharedStringWithSequentialId.insertMarker(0, ReferenceType.Tile, { insert2: true });
+
+        // Process all messages.
+        containerRuntimeFactory.processAllMessages();
+
+        const segmentAt0 = sharedStringWithSequentialId.getContainingSegment(0).segment;
+        const segmentAt1 = sharedStringWithSequentialId.getContainingSegment(1).segment;
+
+        assert(Marker.is(segmentAt0) && Marker.is(segmentAt1), "Make sure we found markers");
+        validateMarker(segmentAt0);
+        validateMarker(segmentAt1);
+        const id0 = segmentAt0.getId();
+        const id1 = segmentAt1.getId();
+        assert(id0 < id1, "Id at 0 should be smaller than id at 1");
+    });
+
+    it("should have a marker id after marker insertion between two markers", async ()=> {
+        sharedStringWithSequentialId.insertMarker(0, ReferenceType.Tile);
+        containerRuntimeFactory.processAllMessages();
```

Comment:
THis is what I am unsure about....I had issues if i didn't process messages after every op, because dds had content of the other ops.

Reply:
With new design this is fixed and there is o need to have a processAllMessages after every insertion